packages/query/src/block-processor.ts

```diff
@@ -361,45 +371,66 @@ export class BlockProcessor implements BlockProcessorInterface {
-    const positionStates: PositionState[] = [
-      new PositionState({ state: PositionState_PositionStateEnum.OPENED }),
-      new PositionState({ state: PositionState_PositionStateEnum.CLOSED }),
-      new PositionState({ state: PositionState_PositionStateEnum.WITHDRAWN, sequence: 0n }),
-    ];
-
+  private async processTransactions(txs: Transaction[]) {
     for (const tx of txs) {
       for (const { action } of tx.body?.actions ?? []) {
-        if (action.case === 'positionOpen' && action.value.position) {
-          for (const state of positionStates) {
-            const metadata = getLpNftMetadata(computePositionId(action.value.position), state);
-            await this.indexedDb.saveAssetsMetadata(metadata);
-          }
-          // to optimize on-chain storage PositionId is not written in the positionOpen action,
-          // but can be computed via hashing of immutable position fields
-          await this.indexedDb.addPosition(
-            computePositionId(action.value.position),
-            action.value.position,
-          );
-        }
-        if (action.case === 'positionClose' && action.value.positionId) {
-          await this.indexedDb.updatePosition(
-            action.value.positionId,
-            new PositionState({ state: PositionState_PositionStateEnum.CLOSED }),
-          );
-        }
-        if (action.case === 'positionWithdraw' && action.value.positionId) {
-          // Record the LPNFT for the current sequence number.
-          const positionState = new PositionState({
-            state: PositionState_PositionStateEnum.WITHDRAWN,
-            sequence: action.value.sequence,
-          });
-          const metadata = getLpNftMetadata(action.value.positionId, positionState);
-          await this.indexedDb.saveAssetsMetadata(metadata);
-
-          await this.indexedDb.updatePosition(action.value.positionId, positionState);
-        }
+        await Promise.all([this.identifyAuctionNfts(action), this.identifyLpNftPositions(action)]);
+      }
+    }
+  }
+
+  private async identifyAuctionNfts(action: Action['action']) {
+    if (action.case === 'actionDutchAuctionSchedule' && action.value.description) {
+      const auctionId = getAuctionId(action.value.description);
+      const metadata = getAuctionNftMetadata(
+        auctionId,
+        // Always a sequence number of 0 when starting a Dutch auction
+        0n,
+      );
+      await this.indexedDb.saveAssetsMetadata(metadata);
+    } else if (action.case === 'actionDutchAuctionEnd' && action.value.auctionId) {
+      const metadata = getAuctionNftMetadata(
+        action.value.auctionId,
+        // Always a sequence number of 1 when ending a Dutch auction
+        1n,
+      );
+      await this.indexedDb.saveAssetsMetadata(metadata);
+    }
+    /**
+     * @todo Handle `actionDutchAuctionWithdraw`, and figure out how to
+     * determine the sequence number if there have been multiple withdrawals.
```

Comment:
I think one way to do that is to structure the `Action['action']` with a sequence number, rather than adding it ex-post in this method, unsure

Reply:
Cool, thanks, added a link to this comment in the ticket (#1020)